src/protocol/check_zero.rs

```diff
@@ -105,9 +105,9 @@ pub mod tests {
                 counter += 1;
 
                 let protocol_output = tokio::try_join!(
```

Comment:
I changed some of these to try_join_all in other places.  Overall, I would prefer we avoid this macro.

Reply:
`try_join_all` allocates [a vector](https://docs.rs/futures-util/0.3.25/src/futures_util/future/try_join_all.rs.html#140) to hold the results, I feel it should be avoided when number of futures is small in prod code. tests don't care